runtime.go

```diff
@@ -278,21 +279,28 @@ func callParamConvert(val reflect.Value, t reflect.Type) reflect.Value {
 
 	switch t.Kind() {
 	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Float32, reflect.Float64:
-		if val.Kind() == t.Kind() {
```

Comment:
Its not clear if you intended to remove this potentially nice optimisation, could you clarify if and why please.